src/control/server/mgmt_pool.go

```diff
@@ -378,11 +394,60 @@ func (svc *mgmtSvc) PoolCreate(ctx context.Context, req *mgmtpb.PoolCreateReq) (
 		return nil, err
 	}
 
-	svc.log.Debugf("MgmtSvc.PoolCreate dispatch resp:%+v\n", resp)
+	svc.log.Debugf("MgmtSvc.PoolCreate dispatch resp:%s\n", mgmtpb.Debug(resp))
 
 	return resp, nil
 }
 
+// checkPools iterates over the list of pools in the system to check
+// for any that are in an unexpected state. Pools not in the Ready
+// state will be cleaned up and removed from the system.
+//
+// NB: Care should be taken to avoid calling this when it could race
+// with a PoolCreate request.
```

Comment:
Hm, this does seem potentially problematic. It seems entirely possible that a leadership transfer could happen during a pool create.

Reply:
If a leadership transfer occurs during a pool create, then the pool will have been left in an indeterminate state and we'll want to clean it up rather than trying to pick up where the last leader left off. At least, I think that's what we want to do, seems pretty hard to get correctly determine the state of a half-created pool with the current infrastructure. We do have some plans to add a fsck-like capability that we could potentially use to figure out what the state of a pool is, but at least for now I think the safest approach is to nuke these when they're found.